examples/compose-replication-semisync.yml

```diff
@@ -0,0 +1,67 @@
+version: "3"
+x-common-variables: &common-variables
+  MARIADB_ROOT_PASSWORD: ${MARIADB_ROOT_PASSWORD}
+  MYSQL_INITDB_SKIP_TZINFO: ${MYSQL_INITDB_SKIP_TZINFO}
+  MARIADB_REPLICATION_USER: ${MARIADB_REPLICATION_USER}
+  MARIADB_REPLICATION_PASSWORD: ${MARIADB_REPLICATION_PASSWORD}
+
+x-common-attributes: &common-attributes
+  image: mariadb:lts
+  env_file:
+    - .env
```

Comment:
in preparation for non-replication based examples here, recommend explicit naming about `env_replication` as a `env_file`.

What's the gain on having it as an env file? It it just showcasing the hiding of passwords out of the main compose file as best practices?